### ISSUE TITLE
[Question] Image Upload over Rest API - Supported Media Type?

### ISSUE THREAD

**Issue by torsten-online** [harvester/harvester#4680]
Hi There - Dear Harvester Community,

I'am writing a python script for Terraform, to provide Image uploads at the initial VM setup (all coded over terraform),
because the terraform provider doesnt really support this, what I want - For example "Harvester Templates".

So the question is - which media type's are supported over the Harvester API?
I used application/octet-stream by using python http requests upload.

I get always the 415er HTTP Error. So, what media type should I use?

Is there any better API Documentation about using image uploads?

I tried this:
https://docs.harvesterhci.io/v1.2/api/create-namespaced-virtual-machine-image

Thanks + have a lot of Fun!
Torsten

**Comment by torsten-online** [comment]
Hi Community, 

I skipped my idea to upload images over python requests script.

Uploading over GUI works fine - but not for image files, like OpenSuSE MicroOS with *.xz Format, because its zipped.

So the workaround is for me now to unpack the image and upload file manually.

Also I defined in terraform a image data source, so the image can be used by the VMs.

Problem there: 

Everytime I get the message, that the image is not available over the name:

`````
data.harvester_image.opensuse_microos_data: Reading...
╷
│ Warning: Empty or non-existent state
│ 
│ There are currently no remote objects tracked in the state, so there is nothing to refresh.
╵
╷
│ Error: virtualmachineimages.harvesterhci.io "opensuse-microos" not found
│ 
│   with data.harvester_image.opensuse_microos_data,
│   on main.tf line 77, in data "harvester_image" "opensuse_microos_data":
│   77: data "harvester_image" "opensuse_microos_data" {
`````

I fixed the problem, by getting the ID of the Image over the GUI / API:
https://harvester.local/v1/harvester/harvesterhci.io.virtualmachineimages/harvester-public/image-XXXX

Using the ID as Name "image-XXXX" fixed the problem for me!

Thanks

Have a lot of Fun!
Torsten
